modules/Bio/EnsEMBL/Hive/DBSQL/AnalysisDataAdaptor.pm

```diff
@@ -81,7 +81,8 @@ sub store_if_needed {
     # UNIQUE KEY in the table definition.
     # This is very similar to check_object_present_in_db_by_content()
     # but it returns the *first* analysis_data_id that's been stored
-    my $sql = 'SELECT MIN(analysis_data_id) FROM analysis_data WHERE md5sum = ? AND data = ?';
+    my $table_name = $self->table_name();
+    my $sql = "SELECT MIN(analysis_data_id) FROM $table_name WHERE md5sum = ? AND data = ?";
```

Comment:
Is it possible that we failed to obtain table name, so request will be failed and we don't process that case, probable need to add error on execute fail?

Reply:
This commit (using `$self->table_name()` in more places) is to fit better the eHive paradigm. The table name is supposed to be defined in `default_table_name` and the queries should be made against `$self->table_name()` whenever possible. This change is not necessary for the purpose of this PR. I may submit it as a separate PR against master, or just remove it